app/publish/publish.go

```diff
@@ -81,7 +82,9 @@ func (h Handler) Handle(w http.ResponseWriter, r *http.Request) {
 }
 
 func publish(sdkAddress, filename string, userID int, qCache cache.QueryCache, rawQuery []byte) []byte {
-	c := query.NewCallerWithCache(sdkAddress, userID, qCache)
+	c := query.NewCaller(sdkAddress, userID)
+	c.Cache = qCache
+	c.DB = boil.GetDB()
```

Comment:
the only way i see of doing this in a middleware is to touch the user anytime they provide their access token, regardless of whether they used the wallet or not. 

i guess i'm fine with this approach, combined with changing the column to `seen_at`. my original intention was to unload unused wallets. this would change that slightly, to unloading wallets of inactive users. that's close enough to the same thing for our purposes that i'm ok with this in exchange for reduced coupling.

Reply:
There's currently no scenario where a user would be on the website and not use his wallet. If not for the fact resolve calls include `wallet_id` now, `wallet_balance` is called every second.